src/Cabinet/Cabinet.cs

```diff
@@ -0,0 +1,207 @@
+        /// </param>
+        /// <param name="dist">
+        /// Number of bytes from origin.
+        /// </param>
+        /// <param name="seektype">
+        /// Initial position.
+        /// </param>
+        /// <returns>
+        /// The offset, in bytes, of the new position from the beginning of the file.
+        /// </returns>
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public delegate uint FNSEEK(int hf, int dist, SeekOrigin seektype);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        public delegate int PFNNOTIFY(NOTIFICATIONTYPE fdint, NOTIFICATION fdin);
+
+        /// <summary>
+        /// The FDICreate function creates an FDI context.
+        /// </summary>
+        /// <param name="pfnalloc">
+        /// Pointer to an application-defined callback function to allocate memory. The function should be declared using the FNALLOC macro.
+        /// </param>
+        /// <param name="pfnfree">
+        /// Pointer to an application-defined callback function to free previously allocated memory. The function should be declared using the FNFREE macro.
+        /// </param>
+        /// <param name="pfnopen">
+        /// Pointer to an application-defined callback function to open a file. The function should be declared using the FNOPEN macro.
+        /// </param>
+        /// <param name="pfnread">
+        /// Pointer to an application-defined callback function to read data from a file. The function should be declared using the FNREAD macro.
+        /// </param>
+        /// <param name="pfnwrite">
+        /// Pointer to an application-defined callback function to write data to a file. The function should be declared using the FNWRITE macro.
+        /// </param>
+        /// <param name="pfnclose">
+        /// Pointer to an application-defined callback function to close a file. The function should be declared using the FNCLOSE macro.
+        /// </param>
+        /// <param name="pfnseek">
+        /// Pointer to an application-defined callback function to move a file pointer to the specified location. The function should be declared using the FNSEEK macro.
+        /// </param>
+        /// <param name="cpuType">
+        /// In the 16-bit version of FDI, specifies the CPU type and can be any of the following values.
+        /// </param>
+        /// <param name="perf">
+        /// Pointer to an ERF structure that receives the error information.
+        /// </param>
+        /// <returns>
+        /// If the function succeeds, it returns a non-<see cref="IntPtr.Zero"/> HFDI context pointer; otherwise, it returns <see cref="IntPtr.Zero"/>.
+        /// </returns>
+        [DllImport(nameof(Cabinet), CallingConvention = CallingConvention.Cdecl)]
+        public static unsafe extern FdiHandle FDICreate(FNALLOC pfnalloc, FNFREE pfnfree, FNOPEN pfnopen, FNREAD pfnread, FNWRITE pfnwrite, FNCLOSE pfnclose, FNSEEK pfnseek, CpuType cpuType, ERF* perf);
```

Comment:
Please add `[Friendly(FriendlyFlags.Out)]` to the `perf` parameter so that an overload with `out ERF` will be generated.
Also: Can you put one parameter per line?